developer/ruby/FindMissingSWIGTypes.rb

```diff
@@ -1,13 +1,91 @@
 require 'json'
+require 'optparse'
 
-# Change this to match your settings
-BUILD_DIRECTORY = '../../build'
+# define the input parameters
+options = Hash.new
+
+optparse = OptionParser.new do |opts|
+
+  opts.on( '-s', '--generated-sources-directory GENERATED_SOURCES_DIRECTORY',
+          String,
+          "Directory when the .cs generated files are, in build/csharp_wrapper/generated_sources") do |sourceDirectory|
+    options[:generatedSourcesDirectory] = sourceDirectory
+  end
+
+  opts.on( '-i', '--ignore-existing-errors', "Ignores the known errors") do |ignoreErrors|
+    options[:ignoreErrors] = ignoreErrors
+  end
+
+  opts.on( '-v', '--verbose', "Print errors that have been ignored") do |verbose|
+    options[:verbose] = verbose
+  end
+end
```

Comment:
FindMissingSwigTypes.rb is now a CLI, without hardcoded paths, so it can be used more easily in a CI context or in Ctest

Reply:
awesome!